### PR TITLE
Documentation on Read the Docs

### PR DESCRIPTION
**Pull request by naved001**
Objective: To host the HIL documentation on [Readthedocs.org](https://readthedocs.org/)

To do that, I formatted the documentation to work with Sphinx and ReadTheDocs (RTD), and added a conf.py and index.srt files required by RTD. 

I published the documentation for demonstration using a branch from my fork, from my account on RTD. It can be viewed [here](http://hil-documentation.readthedocs.io/en/documentation_fix/)

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+Contents:
+
+.. toctree::
+   :maxdepth: 1
+
+   Introduction <README.rst>
+   HaaS Architecture Overview <overview.md>
+   Installation <INSTALL.rst>
+   Installation - Developer <INSTALL-devel.rst>
+   Installation - Developer; TLDR <INSTALL-devel-tldr.rst>
```
Comment (SahilTikale) [inline docs/index.rst]: Rename `TLDR` as `TL;DR`
Put it before the `Installation - Developer`

**Comment by shwsun** [comment]
I would suggest at least providing an entry for rtd.org in `README.rst`. Everything else looks fine to me.

**Comment by SahilTikale** [comment]
make the changes suggested by @shwsun  then sync your fork with the latest changes in the master branch. Let me know if you don't know how to do that.

**Comment by naved001** [comment]
I made the changes suggested by @shwsun and @SahilTikale. 
Can you tell me how to sync my fork with the latest changes in the master? 

Also, the links to the files in docs are broken on [Readthedocs](http://hil-documentation.readthedocs.io/en/documentation_fix/README.html#documentation).

1 way to fix it is to put the complete weblinks in the documentations, that way it will work with Readthedocs and the documentation on github. But then those links will always point to the docs in the master branch regardless of the fork or branch.

**Comment by naved001** [comment]
Made the changes that we discussed. The documentation will now be hosted on MassOpenCloud account at RTD, as soon as this pr is merged (it will generate the documentation from the master branch). It should be available on http://hil.readthedocs.io/en/latest/ 

For demonstration, the latest documentation can be viewed on my account [here](http://hil-documentation.readthedocs.io/en/documentation_fix/index.html)

Let me know if I missed out on something.

**Comment by SahilTikale** [comment]
Merging.

**Pull request merged**